examples/AN00221_freq_domain_example/src/AN00221_freq_domain_example.xc

```diff
@@ -0,0 +1,409 @@
+           //Now the p array is a time domain representation of the selected channel.
+           //   -The imaginary component will be the channel 1 time domain representation.
+           //   -The real component will be the channel 0 time domain representation.
+
+
+           //apply the window function
+           for(unsigned i=0;i<FFT_N/2;i++){
+              long long t = (long long)window[i] * (long long)p[i].re;
+              p[i].re = (t>>31);
+              t = (long long)window[i] * (long long)p[i].im;
+              p[i].im = (t>>31);
+           }
+           for(unsigned i=FFT_N/2;i<FFT_N;i++){
+              long long t = (long long)window[FFT_N-1-i] * (long long)p[i].re;
+              p[i].re = (t>>31);
+              t = (long long)window[FFT_N-1-i] * (long long)p[i].im;
+              p[i].im = (t>>31);
+           }
+
+           for(unsigned i=0; i<FFT_N ; i++) {
+             c_audio <: p[i].re; // output channel 0
+             c_audio <: p[i].im; // output channel 1
+           }
+        }
+    }
+}
+
+
+void output_audio_dbuf_handler(server interface bufget_i input,
+        multichannel_audio_block_s triple_buffer[3],
+        streaming chanend c_audio) {
+
+  unsigned count = 0, sample_idx = 0, buffer_full=0;
+  //unsigned buffer_get_next_bufped_flag=1;
+  int32_t sample;
+
+  unsigned head = 0;          //Keeps track of index of proc_buffer
+
+unsafe {
+  while (1) {
+    // get_next_buf buffers
+    select {
+      case input.get_next_buf(multichannel_audio_block_s * unsafe &buf):
+        // pass ptr to previous buffer
+        if(head==0) {
+          buf = &triple_buffer[NUM_FRAME_BUFFERS-1];
+        } else {
+          buf = &triple_buffer[head-1];
+        }
+        //buffer_get_next_bufped_flag = 1;
+        buffer_full = 0;
+        break;
+
+      // guarded select case will create back pressure.
+      // I.e. c_audio will block until buffer is get_next_bufped
+      case !buffer_full => c_audio :> sample:
+        //if(sample_idx == 0 && !buffer_get_next_bufped_flag) {
+           //printf("Buffer overflow\n");
+        //};
+        unsigned channel_idx = count & 1;
+        if(sample_idx<FFT_N/2) {
+          // first half
+          triple_buffer[head].data[channel_idx][sample_idx] += sample;
+          //buffer->data[channel_idx][sample_idx] = sample;
+        } else {
+          // second half
+          triple_buffer[(head+1)%NUM_FRAME_BUFFERS].data[channel_idx][sample_idx-(FFT_N/2)] = sample;
+        }
+        //buffer->data[channel_idx][sample_idx] = sample;
+        if(channel_idx==1) {
+          sample_idx++;
+        }
+        if(sample_idx>=FFT_N) {
+          sample_idx = 0;
+          buffer_full = 1;
+          //Manage overlapping buffers
+          head++;
+          if(head == NUM_FRAME_BUFFERS)
+              head = 0;
+        }
+
+        count++;
+        break;
+    }
+  }
+}
+}
+
+void init_cs2100(client i2c_master_if i2c){
```

Comment:
It really would be best for this to be replace with the mabs_init_pll() function. That way people stand more of a chance of running it on the wifi board as well in the future.